syntax="proto3";  // 定义版本


// 定义消息，就是定义一种数据结构，用于微服务通讯之间的消息格式约束
message Article {
  // 数字编号就是传输消息的编码后的位置，这是比json效率更高的一种设计，json要传输字段名称过去
  int32 id = 1;
  string title = 2;
  string content = 3;
  string create_time = 4;
}

// 定义rpc请求的入参
message ArticleRequest {
  int32 page = 1;
  int32 page_size = 2;
}

// 定义rpc请求的返回
message ArticleListResponse {
  repeated Article articles = 1;
}

// 文章详情
message ArticleDetailRequest {
  int32 pk = 1;
}
// 定义服务（也就是定义服务和接口）
service ArticleService {
  // 获取文章列表
  rpc List(ArticleRequest) returns (ArticleListResponse);
  // 获取文章详情
  rpc Detail(ArticleDetailRequest) returns (Article);
}